src/widgets/DashboardPage.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2018
+ */
+
+import React, { useState, useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import globalStyles from '../globalStyles';
+import { testData } from '../database/utilities/ReportDemoData';
+import { ReportSideBar } from './ReportSideBar';
+import { ReportChart } from './ReportChart';
+import { createOrUpdateRecord } from '../sync/incomingSyncUtils';
+
+export const DashboardPage = ({ database, settings }) => {
+  const [reports, setReports] = useState(null);
+  const [selectedItemIndex, setSelectedItemIndex] = useState(0);
+
+  // Used for adding data within ./ReportDemoData to the database for
+  // testing while syncing from mobile is not functional.
+  database.write(() =>
+    testData.forEach(report => {
+      const recordCopy = { ...report };
+      recordCopy.data = JSON.stringify(report.data);
+      createOrUpdateRecord(database, settings, 'Report', recordCopy);
+    })
+  );
+
+  useEffect(() => {
+    setReports(
+      database
+        .objects('Report')
+        .snapshot()
+        .map((report, index) => ({
+          id: report.reportID,
+          index,
+          title: report.title,
+          type: report.type,
+          date: report.date,
+          data: JSON.parse(report.data),
+        }))
```

Comment:
I think we don't really need this `map`, doesn't really seem to be do anything!

Reply:
Now that you mentioned that, I see it! Done here: c0f78a3

Reply:
why did you change this to `then`? I don't really understand this? I meant to have this line as just `database.objects("Report")` - what is the `then` doing?